apps/aefate/src/aefa_fate_op.erl

```diff
@@ -600,18 +602,45 @@ map_to_list(Arg0, Arg1, EngineState) ->
     {Map, ES1} = get_op_arg(Arg1, EngineState),
     case Map of
         _ when ?IS_FATE_MAP(Map) ->
-            Tuples = [aeb_fate_data:make_tuple({K, V})
-                      || {K, V} <- maps:to_list(?FATE_MAP_VALUE(Map))],
-            ES2 = write(Arg0, aeb_fate_data:make_list(Tuples), ES1),
+            List = map_to_sorted_list(Map, ES1),
+            ES2 = write(Arg0, List, ES1),
             Size = map_size(?FATE_MAP_VALUE(Map)),
             aefa_engine_state:spend_gas_for_new_cells(Size * 2, ES2);
         ?FATE_STORE_MAP(Cache, MapId) ->
-            {List, ES2} = store_map_to_list(Cache, MapId, ES1),
+            {CleanMap, ES2} = store_map_get_clean(Cache, MapId, ES1),
+            List = map_to_sorted_list(CleanMap, ES2),
             ES3 = write(Arg0, List, ES2),
             Size = length(?FATE_LIST_VALUE(List)),
             aefa_engine_state:spend_gas_for_new_cells(Size * 2, ES3)
     end.
 
+map_to_sorted_list(Map, ES) ->
+    List = maps:to_list(?FATE_MAP_VALUE(Map)),
+    ConsensusVersion = aefa_engine_state:consensus_version(ES),
+    Tuples = if
+        ConsensusVersion < ?IRIS_PROTOCOL_VSN ->
+            MapOrdering = pre_iris_map_ordering(),
+            case maps:get(?FATE_MAP_VALUE(Map), MapOrdering, default) of
+                default ->
+                    [aeb_fate_data:make_tuple(KV)
+                     || KV <- lists:keysort(1, List)];
+                Ordering -> Ordering
+            end;
+        true ->
+            [aeb_fate_data:make_tuple(KV)
+             || KV <- lists:sort(fun ({K1,_}, {K2,_}) -> aeb_fate_data:lt(K1, K2) end, List)]
```

Comment:
We should update the comparison operation to use `aeb_fate_data:lt/2` also in IRIS.